test/builtin/test_datentime.py

```diff
@@ -8,17 +9,28 @@
 
 
 @pytest.mark.skipif(
-    sys.platform in ("win32",) or hasattr(sys, "pyston_version_info"),
+    sys.platform in ("win32",),  # or hasattr(sys, "pyston_version_info"),
+    reason="pyston and win32 does not do well killing threads",
+)
+def test_timeremaining0():
+    str_expr = "TimeConstrained[Integrate[Sin[x]^1000000, x], 0.9]"
+    result = evaluate(str_expr)
+
+    assert result is None or result == Symbol("$Aborted")
```

Comment:
Elsewhere is a suggestion for how to do this in a way were you can be more certain of what the result should be.